pkg/lsp/service.go

```diff
@@ -211,6 +215,39 @@ func (s *ServiceImpl) GetWorkspaceSymbols(ctx context.Context, query string, sym
 	return result, nil
 }
 
+func (s *ServiceImpl) GetDocumentOutline(ctx context.Context, file model.File) (DocumentOutline, error) {
+	project, ok := model.ProjectFromContext(ctx)
+	if !ok {
+		log.Error().Msg("Project not found in context")
+		return DocumentOutline{}, fmt.Errorf("project not found in context")
+	}
+
+	clients := s.getClients(ctx)
+	if len(clients) == 0 {
+		log.Warn().Str("projectId", project.Id).Msg("LSP client not found")
+		return DocumentOutline{}, nil
```

Comment:
we can use `LanguageServerNotFoundError` here

Reply:
Are yo usure we should https://github.com/artmoskvin/hide/blob/9853c8f6284ef075028555b81a9d0e67e3a7f19e/pkg/lsp/service.go#L160

Reply:
Let me explain my thinking here: I think it is better to return explicit errors from LSP service and let the caller decide how to handle them. If you agree we can also fix this in the `GetWorkspaceSymbols` method (not in this PR).

By the way, the `LanguageServerNotFoundError` expects a language ID. In this case, we can either leave it empty or add another error e.g. `LanguageServersNotFoundError`.

DOes this make sense?

Reply:
BTW what is the difference between `LanguageServersNotFoundError`, `LanguageNotSupportedError`

Reply:
the latter is returned when someone tries to start an LSP server for a language that we don't support